tchannel/sync/singleton.py

```diff
@@ -0,0 +1,12 @@
+from __future__ import (
+    absolute_import, division, print_function, unicode_literals
+)
+
+from tchannel.singleton import TChannel as TChannelSingleton
+
+from .client import TChannel as SyncTChannel
+
+
+class TChannel(TChannelSingleton):
+
+    tchannel_cls = SyncTChannel
```

Comment:
Also assign the threadlocal here anew. We don't want sync and async to share the thread local.